### DISCUSSION TITLE
Retention period for blocks_storage

### DISCUSSION THREAD

**Discussion by dyurashevichLT** [cortexproject/cortex#5238] [Q&A]
Hello team, I want configure retention period for blocks_storage
I want collect metrics only 15d
After this period I want delete metrics from FS
How I can setup it ?
Thanks

**Comment by friedrichg** [comment]
```
# Delete blocks containing samples older than the specified retention period. 0
# to disable.
# CLI flag: -compactor.blocks-retention-period
[compactor_blocks_retention_period: <duration> | default = 0s]
```
https://cortexmetrics.io/docs/configuration/configuration-file/#limits_config

Put a flag like this in the compactor

```
 -compactor.blocks-retention-period=15d
```

The retention in the ingesters is controlled by
```
  # TSDB blocks retention in the ingester before a block is removed. This should
  # be larger than the block_ranges_period and large enough to give
  # store-gateways and queriers enough time to discover newly uploaded blocks.
  # CLI flag: -blocks-storage.tsdb.retention-period
  [retention_period: <duration> | default = 6h]
```
It's already 6h. You can increase it to get better query response time, but you need to match it with other flags in compactor and store-gateways.

**Reply by dyurashevichLT** [reply]
Hello, Thanks for your answer
I already have _compactor_blocks_retention_period_
<img width="401" alt="image" src="https://user-images.githubusercontent.com/112411101/228498935-68fce512-1c21-40b8-b5c6-c5733ee233e0.png">
But today, 29 Mar I have metrics from 21 Jan

**Reply by friedrichg** [reply]
check the logs for the compactor, it must be failing to delete them

**Reply by dyurashevichLT** [reply]
cortex log file does not contain any mention of error with compactor 

config part for compactor 
 compactor:
  compaction_interval: 1h
  deletion_delay: 1h
  data_dir: /opt/cortex/compactor
  sharding_ring:
    kvstore:
      store: inmemory
config part for limits
 limits:
  compactor_blocks_retention_period: 62d
  reject_old_samples_max_age: 62d
  max_series_per_metric: 15000000
  max_series_per_query: 10000000
  ingestion_rate: 1000000
  ingestion_burst_size: 200000
   

I don't really understand what is my config mistake

**Comment by ghost** [comment]
facing the same issue. In default config I specified 
_limits:
    compactor_blocks_retention_period: 12h_
and
  _backend: s3
    s3:
      bucket_name: cortex.foo.bar
      region: us-east-1
      endpoint: s3.dualstack.us-east-1.amazonaws.com_
      
Also I have 
_blocks_storage:
  tsdb:
    dir: /tmp/cortex/tsdb
    block_ranges_period:
      - 20m
    retention_period: 30m_

Cortex successfully stored data in S3 but don't want to delete them after 12h. All other options are default from [this file](https://github.com/cortexproject/cortex/blob/master/docs/configuration/single-process-config-blocks.yaml)
What I'm doing wrong?
